python/pyspark/sql/dataframe.py

```diff
@@ -1610,8 +1623,14 @@ def toPandas(self):
         0    2  Alice
         1    5    Bob
         """
-        import pandas as pd
-        return pd.DataFrame.from_records(self.collect(), columns=self.columns)
+        if useArrow:
+            from pyarrow.table import concat_tables
```

Comment:
Since a user might specify useArrow without having the required dependcies installed despite the docstring warning, it would be good to make sure the exception thrown gives the user clear steps to resolve the issue.